lib/IRGen/GenMeta.cpp

```diff
@@ -1782,7 +1791,13 @@ namespace {
       // method for external clients.
       
       // Emit method dispatch thunk.
-      if (hasPublicVisibility(fn.getLinkage(NotForDefinition))) {
+      bool shouldEmitDispatchThunk = false;
```

Comment:
Can this logic, which is the same as the previous change, be extracted into a utility function?

Reply:
It wasn't entirely identical, so I've instead simplified it into just an or-d expression inside the if.